tokenauth/token.go

```diff
@@ -91,12 +101,33 @@ func (t *TokenAuth) CheckToken(token string, purpose string) (*Claims, error) {
 	}
 	kid, _ := parsedToken.Header["kid"].(string)
 	if kid != authServiceReg.PubKey.Kid {
-		return nil, fmt.Errorf("token kid (%s) does not match %s", kid, authServiceReg.PubKey.Kid)
+		refreshed, refreshErr := t.authService.CheckForRefresh()
```

Comment:
This is not quite right. We are still throwing an error above (line 51) if the token was valid. This will prevent the refresh from occurring. We should instead check the token validation error here (if KID does not match) and do a refresh if invalid. 

If the token is valid but the KID doesn't match, then the token must have been generated incorrectly somehow so it should probably be rejected. 

When we check the error from the token validation, we also need to account for the fact that if the token is just expired (valid signature), then we do not want to refresh the keys.